### PR TITLE
fix(occupancy_grid_map_ouliter_filter): reduce radius search region

### PR DESCRIPTION
**Pull request by badai-nguyen**
## Description

In occupancy_grid_map_outlier_filter, the low_confidence_pointcloud extracted from occupancy_grid_map are compared with all input pointcloud. It's waste of time to compare without pointcloud outside of occupancy_grid_map so this PR to exclude pcl outside of occupancy_grid_map from input of RadiusSearch filter. 

## Related links
[TIERIV INTERNAL LINK](https://tier4.atlassian.net/browse/T4PB-25711)

## Tests performed

Before change
Green: high_confidence_pc, Pink: low_confidence_pc
image

2023-02-06-13-25-21_before_reduce_input_pcl_0_pointcloud_number

After change: 
Green: high_confidence_pc, Pink: low_confidence_pc
image

2023-02-06-13-25-21_after_reduce_input_pcl_pointcloud_number

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/occupancy_grid_map_outlier_filter/src/occupancy_grid_map_outlier_filter_nodelet.cpp [hunk 1]
```diff
@@ -230,13 +240,39 @@ void OccupancyGridMapOutlierFilterComponent::onOccupancyGridMapAndPointCloud2(
...
+    if (*x < 0.0) {
+      tmp_behind_pc.push_back(pcl::PointXYZ(*x, *y, *z));
+    } else {
+      tmp_infront_pc.push_back(pcl::PointXYZ(*x, *y, *z));
+    }
+  }
+  pcl::toROSMsg(tmp_infront_pc, input_infront_pc);
+  pcl::toROSMsg(tmp_behind_pc, input_behind_pc);
+  input_infront_pc.header = input_pc->header;
+  input_behind_pc.header = input_pc->header;
```
Comment (yukkysaito) [inline perception/occupancy_grid_map_outlier_filter/src/occupancy_grid_map_outlier_filter_nodelet.cpp:261]: This code would be easier to understand if it were a function.

Comment (badai-nguyen) [reply perception/occupancy_grid_map_outlier_filter/src/occupancy_grid_map_outlier_filter_nodelet.cpp]: @yukkysaito thank you for your review, I changed to function here [9908ca8](https://github.com/autowarefoundation/autoware.universe/pull/2905/commits/9908ca87301ca9f144d3695eb3a2ab06e79f9a90)

File: perception/occupancy_grid_map_outlier_filter/include/occupancy_grid_map_outlier_filter/occupancy_grid_map_outlier_filter_nodelet.hpp [hunk 2]
```diff
@@ -77,7 +78,9 @@ class OccupancyGridMapOutlierFilterComponent : public rclcpp::Node
     const PointCloud2::ConstSharedPtr & input_pointcloud);
   void filterByOccupancyGridMap(
     const OccupancyGrid & occupancy_grid_map, const PointCloud2 & pointcloud,
-    PclPointCloud & high_confidence, PclPointCloud & low_confidence);
+    PclPointCloud & high_confidence, PclPointCloud & low_confidence, PclPointCloud & out_ogm);
+  void splitFontBackPcl(
```
Comment (yukkysaito) [inline perception/occupancy_grid_map_outlier_filter/include/occupancy_grid_map_outlier_filter/occupancy_grid_map_outlier_filter_nodelet.hpp:82]: - Font -> Front
- PCL == PointCloudLibrary 


```suggestion
  void splitPointCloudFrontBack(
```

Comment (badai-nguyen) [reply perception/occupancy_grid_map_outlier_filter/include/occupancy_grid_map_outlier_filter/occupancy_grid_map_outlier_filter_nodelet.hpp]: @yukkysaito I'm sorry for the mistake, I fixed at https://github.com/autowarefoundation/autoware.universe/pull/2905/commits/81ed35f9f686ee2f82470f0188b1911cca256469

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**